tests/credentials_provider_sts_tests.c

```diff
@@ -0,0 +1,717 @@
+    ASSERT_NOT_NULL(provider);
+
+    aws_credentials_provider_get_credentials(provider, s_get_credentials_callback, NULL);
+
+    s_aws_wait_for_credentials_result();
+
+    ASSERT_NOT_NULL(s_tester.credentials);
+    ASSERT_STR_EQUALS("accessKeyIdResp", aws_string_c_str(s_tester.credentials->access_key_id));
+    ASSERT_STR_EQUALS("secretKeyResp", aws_string_c_str(s_tester.credentials->secret_access_key));
+    ASSERT_STR_EQUALS("sessionTokenResp", aws_string_c_str(s_tester.credentials->session_token));
+
+    const char *expected_method = "POST";
+    ASSERT_BIN_ARRAYS_EQUALS(expected_method, strlen(expected_method), s_tester.method.buffer, s_tester.method.len);
+
+    const char *expected_path = "/";
+    ASSERT_BIN_ARRAYS_EQUALS(
+        expected_path, strlen(expected_path), s_tester.request_path.buffer, s_tester.request_path.len);
+
+    ASSERT_TRUE(s_tester.had_auth_header);
+
+    const char *expected_host_header = "sts.amazonaws.com";
+    ASSERT_BIN_ARRAYS_EQUALS(
+        expected_host_header, strlen(expected_host_header), s_tester.host_header.buffer, s_tester.host_header.len);
+
+    ASSERT_BIN_ARRAYS_EQUALS(
+        s_expected_payload.ptr, s_expected_payload.len, s_tester.request_body.buffer, s_tester.request_body.len);
+
+    remove(aws_string_c_str(s_credentials_file_name));
+    aws_credentials_provider_release(provider);
+
+    s_aws_sts_tester_cleanup();
+
+    aws_client_bootstrap_release(bootstrap);
+    aws_host_resolver_clean_up(&resolver);
+    aws_event_loop_group_clean_up(&el_group);
+
+    aws_auth_library_clean_up();
+    return AWS_OP_SUCCESS;
+}
+
+AWS_TEST_CASE(
+    credentials_provider_sts_from_profile_config_succeeds,
+    s_credentials_provider_sts_from_profile_config_succeeds_fn)
+
+static const char *s_env_source_config_file = "[default]\n"
+                                              "aws_access_key_id=BLAHBLAH\n"
+                                              "aws_secret_access_key=BLAHBLAHBLAH\n"
+                                              "\n"
+                                              "[roletest]\n"
+                                              "role_arn=arn:aws:iam::67895:role/test_role\n"
+                                              "credential_source=Environment\n"
+                                              "role_session_name=test_session";
+
+AWS_STRING_FROM_LITERAL(s_env_access_key_val, "EnvAccessKeyId");
+AWS_STRING_FROM_LITERAL(s_env_secret_access_key_val, "EnvSecretAccessKeyId");
+
+static int s_credentials_provider_sts_from_profile_config_environment_succeeds_fn(
+    struct aws_allocator *allocator,
+    void *ctx) {
+    (void)ctx;
+
+    aws_auth_library_init(allocator);
+
+    aws_unset_environment_value(s_default_profile_env_variable_name);
+    aws_unset_environment_value(s_default_config_path_env_variable_name);
+    aws_unset_environment_value(s_default_credentials_path_env_variable_name);
+
+    aws_set_environment_value(s_access_key_id_env_var, s_env_access_key_val);
```

Comment:
We should unset these at the end of this test.

Reply:
process exits.... so....

Reply:
:+1: